src/main/java/net/itsthesky/disky/elements/properties/members/MemberNickname.java

```diff
@@ -50,7 +50,9 @@ public void change(Event e, Object[] delta, Changer.ChangeMode mode, boolean asy
             return;
         final Member member = EasyElement.parseSingle(getExpr(), e, null);
         final String name = (String) delta[0];
```

Comment:
iirc in case of delete/reset, values are empty, thus it'll lead to an exception here anyway. Best thing would be to check before if we're in DELETE or RESET change mode, then either get the name and change it, or set the name to null

Reply:
The only other thing I'd suggest in this case then would be to discourage the usage of `isValid`
general skript convention when it comes down to delta handling would look something like this
```java
final String newNickname = (delta != null && delta[0] instanceOf String) ? (String) delta[0] : null
if (newNickname == null && !EasyElement.equalAny(mode, ChangeMode.RESET, ChangeMode.DELETE))
    return
```

Reply:
It does indeed do a null check, I believe your previous statement of an exception occurring threw me off.
Nothing here should throw an exception as far I saw. (at least using my previous statement)
I do believe `(String) delta[0]` without a null check will 

Although I do still discourage usage of the `isValid` at least in expressions where you have intentional null cases like `reset` and `delete`.

In those expressions at least `isValid` is a somewhat pointless check for delta and leads to false negatives as seen here with `reset` originally not working

Reply:
When using reset or delete, Skript does pass a null value in delta? Because if not, for those modes, it'll be empty, thus `delta[0]` would throw an index out of range, no?

Anyway, I think the best idea would be a mix of yours: having a branch with delete/reset to handle the deletion specifically, then another one to manage the value. Both may store a rest action that'll be queued or not at the end of this big if/else.